dropwizard-jersey/src/main/java/io/dropwizard/jersey/validation/JerseyViolationExceptionMapper.java

```diff
@@ -12,8 +14,14 @@
 
 @Provider
 public class JerseyViolationExceptionMapper implements ExceptionMapper<JerseyViolationException> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JerseyViolationExceptionMapper.class);
+
     @Override
     public Response toResponse(final JerseyViolationException exception) {
+        if (LOGGER.isDebugEnabled()) {
```

Comment:
If you look at the byte code you realize it is necessary. It's splitting hairs for predefined static strings; you are basically saving a one time allocation for the error log string. It gets interesting in the server JVM when branch prediction can optimize away the "if (LOG.isDebugEnabled)" check. I'm not sure LOG.debug itself would be removed in this case; likely you end up still invoking the call and having the method body optimized away (unless your LOG.debug ended up inlined by the JIT).

From a best practice perspective, however, it is ideal to have consistency.

Take this example:

```Java
LOG.debug("Something to say " + result.toString());
```

In this case, what actually happens under the hood is that the toString() method is invoked on result (may or may not be cheap) following by String concatenation (not cheap) before LOG.debug is actually invoked. In the case where `LOG.isDebugEnabled()` would return false, you do a bunch of work for no logging output. This is unlikely to be optimized away as "toString()" varies wildly across implementations and may even have side effects (if it is poorly designed).

If you wrap the call in LOG.isDebugEnabled(), on a cold JVM you pay the cost of a boolean to save some potentially expensive operations. On a hot server JVM the entire block is eliminated. On a hot server JVM in the case debug is true, the if check is simply eliminated.

By always doing this, whether it is needed or not, you ensure that if someone changes the log line to become costly you don't have to worry the developer remembered to add the if {} decoration.

I personally do this for all log levels as a stylistic choice. Typically if you are certain you will always be logging at INFO, it is sufficient to do it only for DEBUG. Since I do turn logging way down from time to time, it makes sense.

Given this is for error handling code and generally shouldn't be hit outside of development or improper handled edge cases (unless you are under attack), if you prefer not having the if {} paradigm for this I will acquiesce as this isn't my codebase.

Thanks!